src/Arrays/MPIStateArrays.jl

```diff
@@ -367,13 +368,33 @@ function LinearAlgebra.norm(Q::MPIStateArray, p::Real=2, weighted::Bool=true)
   else
     locnorm = norm_impl(Q.realQ, Val(p))
   end
+  @show "norm", MPI.Comm_rank(Q.mpicomm), locnorm
 
   mpiop = isfinite(p) ? MPI.SUM : MPI.MAX
   r = MPI.Allreduce([locnorm], mpiop, Q.mpicomm)[1]
+  @show "norm", MPI.Comm_rank(Q.mpicomm), r
   isfinite(p) ? r ^ (1 // p) : r
 end
 
+function normpervar(Q::MPIStateArray, p::Real=2, weighted::Bool=true)
```

Comment:
It might be possible to combine this with the above function by using the same `dims=nothing` trick.

Reply:
Wouldn't this lead to a type instability? (since they return different ranks)